profile.md

```diff
@@ -228,6 +228,25 @@ For example:
 ]
 ```
 
+### JSON Schema Restrictions
+
+Platforms MAY enforce restrictions on the JSON schemas service brokers can use
+to describe the input parameters they accept (see [spec](spec.md#schema-object).
+Each section below will describe any restrictions a Platform places on input
+parameter schemas.
+
+#### Cloud Foundry
+
+The following restrictions are enforced within a Cloud Foundry deployment:
+
+- The `parameters` object MUST have a top-level field as follows:
+  ```
+  "type": "object"
+  ```
+- The `$schema` field MUST be included at the top level of any provided schema.
```

Comment:
Didn't we talk about $schema being optional but then letting the platform pick a default? Or am I mixing up conversations?

Reply:
We did, but in CF we really want to know the exact version so we can tell our validation library which version to validate against. I put this in profile.md to save having this requirement in the core spec, though I expect most broker authors to add the `$schema` key normally anyway.

Reply:
Obviously happy to move both of these to the core spec if everyone agrees with them

Reply:
@duglin do you have any further thoughts on this?

Reply:
I'd prefer for it to be consistent between platforms otherwise we'll have an interop issue, and if we do that then we might as well just put it in the core spec, no?

Reply:
I agree that we want to be consistent here and think we should consider making appropriate changes to the OSB spec...

After reflecting, the feedback I gave earlier around handling multiple spec versions by inference was ultimately not good. Leaving it up to the platform to infer the JSON Schema version is just not of sufficient value to special case in the spec; it only leads to unexpected behaviour.  So... IF an author would like their parameters to be validated according to JSON Schema the contents of "parameters" MUST be a valid JSON Schema document (with a valid "$schema" property value that the platform understands) or else be empty.

Beyond stating that the document must be a valid JSON Schema document I would not stipulate "type": "object" on the schema as this requirement will vary with the schema version. e.g. in v4 only "type": "object" is permitted but in v6 the meta-scheme permits "type": "boolean, object", and who knows what future versions might say here. I would leave this detail up to the schema spec and not in OSB.